packages/hash/frontend/src/pages/[shortname]/types/entity-type/[entity-type-id].page/use-entity-type-value.tsx

```diff
@@ -19,10 +25,47 @@ import {
   useFetchEntityTypes,
 } from "../../../../../shared/entity-types-context/hooks";
 
+interface EntityTypeAndPropertyTypes {
+  entityType: EntityType;
+  propertyTypes: Record<string, PropertyType>;
+}
+
+const getPropertTypes = (
+  properties: any,
+  subgraph: Subgraph,
+  propertyTypes?: Map<string, PropertyType>,
+) => {
+  let propertyTypesMap = propertyTypes ?? new Map<string, PropertyType>();
+  for (const prop of properties) {
+    const propertyUri = "items" in prop ? prop.items.$ref : prop.$ref;
+    if (!propertyTypesMap.has(propertyUri)) {
+      const propertyType = getPropertyTypeById(subgraph, propertyUri)?.schema;
+
+      if (propertyType) {
+        for (const childProp of propertyType.oneOf) {
+          if ("type" in childProp && childProp.type === "object") {
+            propertyTypesMap = getPropertTypes(
+              Object.values(childProp.properties),
+              subgraph,
+              propertyTypesMap,
+            );
+          }
+        }
+      }
+
+      if (propertyType) {
+        propertyTypesMap.set(propertyUri, propertyType);
+      }
+    }
+  }
+
+  return propertyTypesMap;
+};
+
 export const useEntityTypeValue = (
```

Comment:
I think we're approaching (or at) the point where it'd be better to just have a subgraph we can pass around and extract things from as needed, since we already have helper utilities to do so – but I appreciate that might involve a lot of changes, and be incompatible with how we currently store draft state.

Reply:
In the end we just need the property types relativant to the entity type, which we get here. Is there something else I'm missing?

Reply:
I was just thinking that we do some extraction of property types from a subgraph here, and then do more manipulation / extraction of them elsewhere, and maybe we could reduce some of it. But maybe we can't – it's fine as it is